node/pkg/txverifier/utils.go

```
@@ -26,6 +27,69 @@ const (
 	CacheDeleteCount = 10
 )
 
+// msgID is a unique identifier the corresponds to a VAA's "message ID".
+// i.e.emitter_chain/emitter_address/sequence tuple. In this package, it is used
+// to identify a single LogMessagePublished event, which in turn maps onto a unique
+// MessagePublication elsewhere in the Guardian code.
+type msgID struct {
+	// Sequence of the VAA
+	Sequence uint64
+	// EmitterChain the VAA was emitted on
+	EmitterChain vaa.ChainID
+	// EmitterAddress of the contract that emitted the Message
+	EmitterAddress vaa.Address
+}
+
+// MessageID returns a human-readable emitter_chain/emitter_address/sequence tuple.
+func (m *msgID) String() string {
+	return fmt.Sprintf("%d/%s/%d", m.EmitterChain, m.EmitterAddress, m.Sequence)
+}
+
+func (m *msgID) Empty() bool {
+	return m.EmitterChain == 0 && m.EmitterAddress == ZERO_ADDRESS_VAA && m.Sequence == 0
+}
+
+// NewMsgID creates a new msgID from a string in the format "chainID/emitterAddress/sequence".
+func NewMsgID(in string) (msgID, error) {
+	if len(in) == 0 {
+		return msgID{}, errors.New("msgIDStr is empty")
+	}
+	parts := strings.Split(in, "/")
+	if len(parts) != 3 {
+		return msgID{}, errors.New("invalid msgID: must be in the format chainID/emitterAddress/sequence")
+	}
+
+	chainID, err := vaa.StringToKnownChainID(parts[0])
+	if err != nil {
+		return msgID{}, err
+	}
+
+	supported := IsSupported(chainID)
+	if !supported {
+		return msgID{}, err
```

If we return here, the error is going to be `nil`. Is this intentional (return empty msgID + nil error), or did you want to return an actual error here?

---

nice catch, no this should be an error, not nil.